rpc/src/eth.rs

```diff
@@ -246,6 +253,39 @@ impl<B, C, P, CT, BE, A> EthApi<B, C, P, CT, BE, A> where
 		);
 		(best_number, header_number)
 	}
+
+	fn current_block(&self, id: &BlockId<B>) -> Option<ethereum::Block> {
+		if let Ok(Some(block_data)) = self.client.storage(
+			&id,
+			&StorageKey(
+				storage_prefix_build(b"Ethereum", b"CurrentBlock")
+			)
+		) {
+			return Some(Decode::decode(&mut &block_data.0[..]).unwrap());
+		} else { return None; };
```

Comment:
This is hard to read, and I'm not sure about the error handling. I think a match statement might helps. What does it mean when you get an `Err` vs an `Ok(None))`? Is it safe to unwrap here? If so, why not use `expect`?

Reply:
> This is hard to read, and I'm not sure about the error handling. I think a match statement might helps.


I prefer to use if let when there are only two possible values to match.

> What does it mean when you get an Err vs an Ok(None))


Error can be https://crates.parity.io/sp_blockchain/enum.Error.html, and Ok(None) the storage value was not found for that StorageKey at that block height.

> Is it safe to unwrap here?


Yeah, Decode might return an error. Will fix!